src/main/java/frc/robot/subsystems/BackClimber.java

```diff
@@ -5,58 +5,109 @@
 
-  /** Creates a new Climber. */
+  /**
+   * Creates a new Climber.
+   */
   public BackClimber() {
+    super(new PIDController(kP, kI, kD));
+    getController().setIntegratorRange(0, 0.2);
+
     m_backClimber = new WPI_TalonFX(Constants.kBackClimberTalonId, Constants.kCanivoreName);
     m_backCanCoder = new CANCoder(Constants.kBackCanCoderId, Constants.kCanivoreName);
 
-  
 
     m_backCanCoder.configFactoryDefault();
     m_backCanCoder.configMagnetOffset(Constants.kBackClimberCanCoderOffset);
     m_backCanCoder.configAbsoluteSensorRange(AbsoluteSensorRange.Signed_PlusMinus180);
     m_backCanCoder.configSensorInitializationStrategy(SensorInitializationStrategy.BootToAbsolutePosition);
-
+    m_backCanCoder.configSensorDirection(true);
 
     m_backClimber.configFactoryDefault();
     m_backClimber.configForwardSoftLimitEnable(true);
     m_backClimber.configReverseSoftLimitEnable(true);
     m_backClimber.configReverseSoftLimitThreshold(Constants.kBackClimberReverseLimit);
     m_backClimber.configForwardSoftLimitThreshold(Constants.kBackClimberForwardLimit);
+    m_backClimber.setNeutralMode(NeutralMode.Brake);
 
     m_backClimber.configRemoteFeedbackFilter(m_backCanCoder, 0);
     m_backClimber.configSelectedFeedbackSensor(RemoteFeedbackDevice.RemoteSensor0);
+    m_backClimber.setSensorPhase(true);
+
+    m_backClimber.setInverted(true);
 
   }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
+    super.periodic();
     SmartDashboard.putNumber("Climber Back Angle", m_backCanCoder.getPosition());
     SmartDashboard.putNumber("Climber Back Absolute Angle", m_backCanCoder.getAbsolutePosition());
     SmartDashboard.putNumber("Climber Back Selected Sensor position", m_backClimber.getSelectedSensorPosition());
+    SmartDashboard.putNumber("Back Climber Stator Current", m_backClimber.getStatorCurrent());
+    SmartDashboard.putNumber("Back Climber Supply Currnet", m_backClimber.getSupplyCurrent());
+    SmartDashboard.putNumber("Climber Back Error", getController().getPositionError());
   }
 
   public void startClimbMotor(int direction, double speed) {
-    m_backClimber.set(ControlMode.PercentOutput, direction*speed);
+    m_backClimber.set(ControlMode.PercentOutput, direction * speed);
+  }
+
+  // TODO: Adjust to match hardware
+  public void startOutwardClimb() {
+    m_backClimber.set(ControlMode.PercentOutput, Constants.kBackClimberSpeed);
+  }
+
+  public void startInwardClimb() {
+    m_backClimber.set(ControlMode.PercentOutput, -Constants.kBackClimberSpeed);
+  }
+
+  public Rotation2d getAngle() {
+    return Rotation2d.fromDegrees(this.m_backCanCoder.getAbsolutePosition());
   }
 
-  public void stopClimbMotor() {
+  public void stop() {
     m_backClimber.neutralOutput();
   }
+
+  @Override
+  protected void useOutput(double output, double setpoint) {
+    // TODO Auto-generated method stub
+
+    if(output>0.1){
+      output = 0.1;

```

Comment:
This value should get be increased so we have can have back climber arms to move at a reasonable speed. Right now its kind of slow. We want it slow/fast enough so we can gracefully move the robot into position to get the front arms onto the next bar.